src/usage.rs

```diff
@@ -1,56 +1,51 @@
 use crate::ToParams;
 
 #[derive(Debug, Clone)]
-pub struct Usage {
-    metrics: Vec<(String, String)>,
+pub struct MetricUsage<'m>(&'m str, &'m str);
+
+impl<'m, M: AsRef<str> + 'm, V: AsRef<str> + 'm> From<&'m (M, V)> for MetricUsage<'m> {
+    fn from((m, v): &'m (M, V)) -> Self {
+        Self(m.as_ref(), v.as_ref())
+    }
 }
 
-// TODO fix this to stop the wild cloning of strings
-// IMO we should treat Usage as a container itself and share it in multiple calls
-// ie.
-// let u = Usage::new();
-// u.push(("metric1", 10));
-// u.push(("metric2", 20));
-// We should also be able to cache the generated parameters, and turn into the inner container,
-// clone, iterate, etc.
-// An internal Vec should do, as we are unlikely to need direct access to specific entries once we
-// have a usage (we can do it anyway, but searching sequentially). If we needed something more
-// powerful we can always write conversions from Vec/HashMap to Usage.
-impl Usage {
-    /// Creates a `Usage`.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use threescalers::usage::*;
-    ///
-    /// let mut metrics = Vec::new();
-    /// metrics.push(("metric1", 10));
-    /// metrics.push(("metric2", 20));
-    /// let usage = Usage::new(&metrics);
```

Comment:
It would be nice to keep these doc-tests.

Reply:
My bad, will recreate these